@tgamblin 

We don't know the exact issue but here is a reference error during the ci build

```
zuznj55642V9eePfuWMOScFFGPdZldRdlyf10DVpGhNT81o6c79H9zB0Y7+IYJgFesJgOxvjJm008IwrXeSNsT8nJRXUL0Mz1YoaVGP5B1GZla+kfKHzBL4zlOg7iaRiKCN8cGFI46HLG+7rdoADpCir+qlf7ojkYUTTr3wO25i7lLFfx+xs+xhXaaOSZ1Zuc7rki5xOxbLX8S7O3vOvm1wTgxtp4WI/DtY60lWu5VWiK9HmtjJyNsI+l1rGP8wu8SztK9AYVLQjs3IYVMVMmTWyQanu2XMs9O9OXLMlSCCyLJxeIm67Ki6zeEiRULQp8wHcHzsyqK4ndRwjXA9FxaQBFOjeGqVZ6D1O32dOt3c/26KC2wlKM2kjuqd8JlvrafZk/U/leL3Kq9C5w6vEiifyz68fmbOF/Us56U/R4OKzNo8K7Nc0ON8aPh2C7Qzxq7QB79XX/f/dNtdNN0+rME7rvGKhs2BQcWvqKi6S3GT2/nM5D/2Y/wPE/2cL
==> [2020-07-08-11:11:34.063114, 65055] remote_mirror_url = s3://cori
==> [2020-07-08-11:11:34.063146, 65055] job_spec_pkg_name = openblas
==> [2020-07-08-11:11:34.063172, 65055] compiler_action = NONE
==> [2020-07-08-11:11:34.065690, 65055] Current working directory: /global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental, Contents:
==> [2020-07-08-11:11:34.066433, 65055]   .gitlab-ci.yml
==> [2020-07-08-11:11:34.066484, 65055]   .git
==> [2020-07-08-11:11:34.066517, 65055]   jobs_scratch_dir
==> [2020-07-08-11:11:34.066545, 65055]   .spack-env
==> [2020-07-08-11:11:34.066571, 65055]   artifacts
==> [2020-07-08-11:11:34.066595, 65055]   spack
==> [2020-07-08-11:11:34.066620, 65055]   spack.yaml
==> [2020-07-08-11:11:34.068386, 65055] job concrete spec path: /global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/jobs_scratch_dir/specs/openblas.yaml
==> [2020-07-08-11:11:34.068439, 65055] ci.import_signing_key() will attempt to import a key
==> [2020-07-08-11:11:34.144905, 65055] spack gpg list:
==> [2020-07-08-11:11:34.145069, 65055] gpg: keybox '/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/opt/spack/gpg/pubring.kbx' created
gpg: /global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/opt/spack/gpg/trustdb.gpg: trustdb created

==> [2020-07-08-11:11:44.226784, 65055] ProcessError: Command exited with status 2:
    '/usr/bin/gpg2' '--import' '/tmp/tmp3m3nts2u/signing_key'
==> [2020-07-08-11:11:44.227195, 65055] ProcessError: Command exited with status 2:
    '/usr/bin/gpg2' '--import' '/tmp/tmp3m3nts2u/signing_key'
==> [2020-07-08-11:11:44.227262, 65055] Error: Command exited with status 2:
'/usr/bin/gpg2' '--import' '/tmp/tmp3m3nts2u/signing_key'
Traceback (most recent call last):
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/main.py", line 761, in main
    return _invoke_command(command, parser, args, unknown)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/main.py", line 489, in _invoke_command
    return_val = command(parser, args)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/cmd/ci.py", line 386, in ci
    args.func(args)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/cmd/ci.py", line 251, in ci_rebuild
    spack_ci.import_signing_key(signing_key)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/ci.py", line 837, in import_signing_key
    key_import_output = spack_gpg('trust', sign_key_path, output=str)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/main.py", line 545, in __call__
    self.command, self.parser, args, unknown)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/main.py", line 489, in _invoke_command
    return_val = command(parser, args)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/cmd/gpg.py", line 163, in gpg
    args.func(args)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/cmd/gpg.py", line 131, in gpg_trust
    Gpg.trust(args.keyfile)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/util/gpg.py", line 101, in trust
    cls.gpg()('--import', keyfile)
  File "/global/cscratch1/ecp-gitlab/pipelines/experimental/builds/users/lpeyrala/rF54Q1bH/0/pipelines/experimental/spack/lib/spack/spack/util/executable.py", line 189, in __call__
    proc.returncode, long_msg)
spack.util.executable.ProcessError: Command exited with status 2:
    '/usr/bin/gpg2' '--import' '/tmp/tmp3m3nts2u/signing_key'
```

For now we are using Luke fork spack repo with pipeline and once this change is approved we can switch to upstream develop branch.  You can see the latest pipeline build https://software.nersc.gov/pipelines/e4s/-/pipelines/5116 are progressing, before we couldn't get anything built.

--- a/lib/spack/spack/util/gpg.py
+++ b/lib/spack/spack/util/gpg.py
@@ -13,7 +13,7 @@ from spack.util.executable import which
 
 _gnupg_version_re = r"^gpg \(GnuPG\) (.*)$"
 
-GNUPGHOME = spack.paths.gpg_path
+GNUPGHOME = os.getenv('SPACK_GNUPGHOME', spack.paths.gpg_path)
 
 
 def parse_keys_output(output):

